src/System.Runtime.Environment/src/OSName.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Runtime.InteropServices
+{
+    public struct OSName : IEquatable<OSName>
+    {
+        private readonly string _name;
+
+        private const string WindowsName = "WINDOWS";
+        private const string LinuxName = "LINUX";
+        private const string OSXName = "OSX";
+
+        public static readonly OSName Windows = new OSName(WindowsName);
+        public static readonly OSName Linux = new OSName(LinuxName);
+        public static readonly OSName OSX = new OSName(OSXName);
+
+        public OSName(string name)
+        {
+            if (name == null) throw new ArgumentNullException("name");
+            if (name.Length == 0) throw new ArgumentException(SR.Format(SR.Argument_EmptyValue, "name"));
```

Comment:
Why not: `throw new ArgumentException(SR.Argument_EmptyValue, "name");`, with the resource simply as `"Value cannot be empty."`?

Passing the parameter name to `ArgumentException`'s constructor overload that takes a `paramName` parameter will take care of adding the `paramName` to the message (and makes it available via `ArgumentException.ParamName`); no need to use `SR.Format` here.